### DISCUSSION TITLE
PID tuning

### DISCUSSION THREAD

**Discussion by deividAlfa** [deividAlfa/stm32_soldering_iron_controller#79] [General]
Here you can show your PID settings for others to try.
I'll post my current settings as example:

**PID settings**
> Kp 36.00
Ki 65.00
Kd 5.00
Imax 0.60
Imin 0.00

**TIP**
> T12-D12
Also works very well with most of my tips.

**IRON**
If not default: ADC Time, Delay and filter settings (System response is very sensitive to this).
> ADC time 100ms
ADC Delay 10.0ms
Filtering: Default
 
**SYSTEM**
> Gui Delay 100ms (To update the screen faster, doesn't affect the PID or tip performance)

**Comment by sst311212** [comment]
**PID**
> Kp: 35.00
> Ki: 25.00
> Kd: 5.00
> Imax: 0.65
> Imin: 0.00

**TIP**
> T12-D24
> C245-911

**IRON**
> ADC Time: 100ms
> ADC Delay: 10.0ms
> PWM Mul: 10x (For C245)
> Filtering: Default

Tried 65 Ki, but temperature start swinging while doing coin(3x) test.
You can raise Ki and Imax a little bit if using cartridge is larger than C245-911.
Both cartridges are genuine, clone one won't transfer heat such fast, thus system doesn't know temperature is dropping.
